spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+RSpec.describe 'Customer Show Page', type: :feature do
+  describe 'As a visitor' do
+    before(:each) do
+      @supermarket_1 = Supermarket.create!(name: "Ralphs", location: "Over There St.")
+      @customer_1 = Customer.create!(name: "Alex Long")
+      @item_1 = @supermarket_1.items.create!(name: "Apple", price: 5)
+      @item_2 = @supermarket_1.items.create!(name: "Banana", price: 2)
+      @item_3 = @supermarket_1.items.create!(name: "Orange", price: 3)
+      @item_4 = @supermarket_1.items.create!(name: "Onion", price: 1)
+      @item_5 = @supermarket_1.items.create!(name: "Bell Pepper", price: 4)
+      @customer_item_1 = CustomerItem.create!(customer_id: @customer_1.id, item_id: @item_1.id)
+      @customer_item_2 = CustomerItem.create!(customer_id: @customer_1.id, item_id: @item_2.id)
+      @customer_item_3 = CustomerItem.create!(customer_id: @customer_1.id, item_id: @item_3.id)
+    end
+
+    it 'shows the customers name and list of its items' do
+      visit "/customers/#{@customer_1.id}"
+
+      expect(page).to have_content(@customer_1.name)
+      expect(page).to have_content(@item_1.name)
+      expect(page).to have_content(@item_2.name)
+      expect(page).to have_content(@item_3.name)
+    end
+
+    it 'shows the price of each item and the supermarket it belongs to' do
+      visit "/customers/#{@customer_1.id}"
+
+      within "#item-#{@item_1.id}" do
+        expect(page).to have_content(@item_1.price)
+        expect(page).to have_content(@supermarket_1.name)
+      end
+      within "#item-#{@item_2.id}" do
+        expect(page).to have_content(@item_2.price)
+        expect(page).to have_content(@supermarket_1.name)
+      end
+      within "#item-#{@item_3.id}" do
+        expect(page).to have_content(@item_3.price)
+        expect(page).to have_content(@supermarket_1.name)
+      end
+    end
+
+    it 'has a form to add an item to the customer' do
+      visit "/customers/#{@customer_1.id}"
+
+      within '#add_item' do
+        expect(page).to have_content("Add Item to Customer")
+        expect(page).to have_field("item_id")
+        expect(page).to have_button("Submit")
+      end
+    end
+
+    it 'can add an item to the customer by item id number' do
+      visit "/customers/#{@customer_1.id}"
+
+      fill_in "item_id", with: "#{@item_4.id}"
+      click_on "Submit"
+
+      expect(current_path).to eq("/customers/#{@customer_1.id}")
+
+      within "#item-#{@item_4.id}" do
+        expect(page).to have_content(@item_4.name)
+        expect(page).to have_content(@item_4.price)
+        expect(page).to have_content(@supermarket_1.name)
+      end
+    end
```

Comment:
This test is great, one recommendation I would make is, before you fill out the form and submit, do a check to make sure the item you're adding IS NOT already there:

```ruby
within "#item-#{@item_4.id}" do
  expect(page).to_not have_content(@item_4.name)
end

# The rest of the form stuff

within "#item-#{@item_4.id}" do
  expect(page).to have_content(@item_4.name)
  expect(page).to have_content(@item_4.price)
  expect(page).to have_content(@supermarket_1.name)
end
```

Now this way, we know that the addition of `@item_4` is actually from the form.